### PR TITLE
feat: add stray token detection in if/in qualifier conditions

### PR DESCRIPTION
**Pull request by jbearak**
## Summary

Adds diagnostic detection for stray tokens that appear after comparison expressions in `if` and `in` qualifier conditions. For example, `if (x == y oops)` or `if x == y oops` now emit a diagnostic warning on the unexpected token.

### New Error Codes
- `STRAY_TOKEN_IN_CONDITION = 3013` - for unexpected tokens after comparisons
- `SPLIT_LITERAL_IN_CONDITION = 3014` - for split literal patterns like `. 9`

### Parser Enhancements
- State machine in `parseIfQualifierExpression()` and `parseInQualifierExpression()` with states: INITIAL, AFTER_OPERAND, AFTER_COMPARE, AFTER_RHS
- Helper methods: `isComparisonOperator()`, `isLogicalOperator()`, `isArithmeticOperator()`, `isValidAfterComparison()`
- `detectSplitLiteral()` for patterns like `. 9` → `.9`, `. a` → `.a` (extended missing)

### Features
- Detects stray tokens after all comparison operators (`==`, `!=`, `~=`, `<`, `>`, `<=`, `>=`)
- Handles nested parentheses correctly
- Handles negation operators (`!`, `~`)
- Works across continuation lines (`///`)
- No false positives for valid compound expressions with `&`, `|`, arithmetic operators

### Tests
- 35 unit tests covering basic cases, edge cases, and diagnostic quality
- 8 property-based tests with 100 iterations each

## Spec
See `.kiro/specs/stray-token-in-condition/` for requirements, design, and task list.

## Summary by CodeRabbit

* **New Features**
  * Detect stray tokens after comparison expressions in if/in qualifier conditions and surface actionable diagnostics.
  * Detect split numeric/string literal issues in conditions and suggest fixes.
  * Recognize extended missing-value forms (dot+letter) to avoid false positive split-literal diagnostics.

* **Documentation**
  * Added design, requirements, and task specs detailing behavior, edge cases, and acceptance criteria.

* **Tests**
  * Added comprehensive unit and property tests for stray-token, split-literal, continuation, and tokenization scenarios.

* **Chores**
  * Updated diagnostics extraction so new diagnostics are surfaced in tooling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -330,6 +330,8 @@ export enum ParseErrorCode {
   MISSING_EXPRESSION_AFTER_EQUALS = 3010,
   UNBALANCED_PARENTHESES = 3011,
   ORPHAN_CLOSE_BRACE = 3012,
+  STRAY_TOKEN_IN_CONDITION = 3013,
+  SPLIT_LITERAL_IN_CONDITION = 3014,
 }
```
Comment (jbearak) [inline src/types/index.ts]: We need to update the codebase to know that these error codes now end at 3014, not 3012; look for and fix the `<= 3012` condition (we should revise that conditional)

**Comment by jbearak** [comment]
/q can u not read the PR comments?
